### PR TITLE
hv: nested: multiple active VMCS12 - part 2

### PR DESCRIPTION
**Pull request by ZideChen0**

### CODE REVIEW

File: hypervisor/include/arch/x86/asm/guest/nested.h [hunk 1]
```diff
@@ -333,12 +334,20 @@ struct acrn_vvmcs {
...
 	bool host_state_dirty;		/* To indicate need to merge VMCS12 host-state fields to VMCS01 */
 	bool control_fields_dirty;	/* For all other non-host-state fields that need to be merged */
 } __aligned(PAGE_SIZE);
 
+/*
+ * MAX_ACTIVE_VVMCS_NUM could be of any arbitrary number. We chose MAX_VCPUS_PER_VM
+ * because it is large enough so that vvmcs[] is able to cache most active VMCS12s
+ * in most configurations.
+ */
+#define MAX_ACTIVE_VVMCS_NUM	MAX_VCPUS_PER_VM
```
Comment (dongyaozu) [inline hypervisor/include/arch/x86/asm/guest/nested.h:347]: Don't link this to MAX_VCPUS_PER_VM. It is completely seperate.

**Review by dongyaozu — CHANGES_REQUESTED** [review]

File: hypervisor/include/arch/x86/asm/guest/nested.h (hunk 1 above)
Comment (ZideChen0) [reply hypervisor/include/arch/x86/asm/guest/nested.h]: > Don't link this to MAX_VCPUS_PER_VM. It is completely seperate.

OK. I changed MAX_ACTIVE_VVMCS_NUM to 4.

**Comment by ZideChen0** [comment]
The size of nept_desc_bucket[] is directly related to the total number of vvmcs[]. In [patch 1/2] added this change:

- #define CONFIG_MAX_GUEST_EPT_NUM	4
+ #define CONFIG_MAX_GUEST_EPT_NUM	(MAX_ACTIVE_VVMCS_NUM * MAX_VCPUS_PER_VM)

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**